src/App.tsx

```diff
@@ -17,18 +65,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                filterByStatus={filterByStatus}
+                setFilterByStatus={setFilterByStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isListLoading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodoId={selectedTodoId}
+                  setSelectedTodoId={setSelectedTodoId}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {(selectedTodoId !== 0) && (
+        <TodoModal
+          todos={todos}
+          selectedTodoId={selectedTodoId}
+          isModalLoading={isModalLoading}
+          setIsModalLoading={setIsModalLoading}
```

Comment:
`const [isModalLoading, setIsModalLoading] = useState(false);`
This state you should keep inside the modal as it's local and don't use it here